fuels-abigen-macro/tests/harness.rs

```diff
@@ -1195,3 +1208,104 @@ async fn test_contract_calling_contract() {
 
     assert!(!res.value);
 }
+
+#[tokio::test]
+async fn test_gas_errors() {
+    let mut rng = StdRng::seed_from_u64(2322u64);
+
+    // Generates the bindings from the an ABI definition inline.
+    // The generated bindings can be accessed through `MyContract`.
+    abigen!(
+        MyContract,
+        "fuels-abigen-macro/tests/test_projects/contract_test/out/debug/contract_test-abi.json"
+    );
+
+    // Build the contract
+    let salt: [u8; 32] = rng.gen();
+    let salt = Salt::from(salt);
+
+    let compiled = Contract::load_sway_contract(
+        "tests/test_projects/contract_test/out/debug/contract_test.bin",
+        salt,
+    )
+    .unwrap();
+
+    let (provider, wallet) = setup_test_provider_and_wallet().await;
+    let contract_id = Contract::deploy(
+        &compiled,
+        &provider.clone(),
+        &wallet,
+        TxParameters::default(),
+    )
+    .await
+    .unwrap();
+
+    let contract_instance = MyContract::new(contract_id.to_string(), provider, wallet);
+
+    // Test for insufficient gas.
+    let result = contract_instance
+        .initialize_counter(42) // Build the ABI call
+        .tx_params(TxParameters::new(Some(1_000), Some(100), None))
+        .call() // Perform the network call
+        .await
+        .expect_err("should error");
+
+    assert_eq!("Contract call error: Response errors; unexpected block execution error InsufficientGas { provided: 1, required: 100000 }", result.to_string());
+
+    // Test for running out of gas. Gas price as `None` will be 0.
+    // Gas limit will be 100, this call will use more than 100 gas.
+    let result = contract_instance
+        .initialize_counter(42) // Build the ABI call
+        .tx_params(TxParameters::new(None, Some(100), None))
+        .call() // Perform the network call
+        .await
+        .expect_err("should error");
+
+    assert_eq!("Contract call error: OutOfGas", result.to_string());
+}
+
+#[tokio::test]
+async fn token_ops() {
+    abigen!(
+        TestFuelCoinContract,
+        "fuels-abigen-macro/tests/test_projects/token-ops/out/debug/token-ops-abi.json"
+    );
+
+    let salt = Salt::from([0u8; 32]);
+    let compiled = Contract::load_sway_contract(
+        "tests/test_projects/token-ops/out/debug/token-ops.bin",
+        salt,
+    )
+    .unwrap();
+
+    let (provider, wallet) = setup_test_provider_and_wallet().await;
+
+    let id = Contract::deploy(&compiled, &provider, &wallet, TxParameters::default())
+        .await
+        .unwrap();
+
+    let instance = TestFuelCoinContract::new(id.to_string(), provider.clone(), wallet.clone());
+
+    let target = testfuelcoincontract_mod::ContractId { value: id.into() };
```

Comment:
what's `testfuelcoincontract_mod`

Reply:
The generated module/namespace for potentially conflicting names, such as ContractID (which is created by the user in this case here). The name is based on the contract/project name. See sway-lib-std's token-ops tests for more context/rationale.